### PR TITLE
[language-platform] move uploads background jobs to its own layer

### PR DESCRIPTION
**Pull request by Numbers88s**
Move uploads background jobs to its own layer

## Test plan
Updated and ran all the test.

### CODE REVIEW

File: internal/codeintel/uploads/service.go [hunk 1]
```diff
@@ -320,3 +384,247 @@ func (s *Service) GetListTags(ctx context.Context, repo api.RepoName, commitObjs
...
+}
+
+func (s *Service) SoftDeleteExpiredUploads(ctx context.Context) (int, error) {
+	ctx, _, endObservation := s.operations.softDeleteExpiredUploads.With(ctx, nil, observation.Args{})
+	defer endObservation(1, observation.Args{})
+
+	return s.store.SoftDeleteExpiredUploads(ctx)
+}
+
+// BackfillCommittedAtBatch calculates the committed_at value for a batch of upload records that do not have
```
Comment (efritz) [inline internal/codeintel/uploads/service.go:454]: I think the update [here](https://github.com/sourcegraph/sourcegraph/pull/43010/files#diff-1be0b331307b4872347c6dfa9654b7a6f07d088f145a6a9b64a0749d049e75f3) got a bit clobbered.

I think this comment is the only casualty, but please check to make sure the rest of the changes in that PR are still in effect.

**Review by efritz — APPROVED** [review]

File: internal/codeintel/uploads/service.go (hunk 1 above)
Comment (Numbers88s) [reply internal/codeintel/uploads/service.go:454]: Yeah I'll check it out first. Not merging till I'm sure. Thanks for pointing it out.

File: internal/codeintel/uploads/service_backfill_test.go [hunk 2]
```diff
@@ -88,7 +98,17 @@ func TestBackfillCommittedAtBatchUnknownCommits(t *testing.T) {
...
+		store,
+		nil, // repoStore
+		nil, // lsifstore
+		gitserverClient,
+		nil, // policySvc
+		nil, // policyMatcher
+		nil, // locker
+		nil, // backgroundJobs
+		&observation.TestContext,
+	)
```
Comment (chrismwendt) [inline internal/codeintel/uploads/service_backfill_test.go:111]: If this gets unruly later on, sometimes I've seen long param lists get converted to a struct like `type newServiceArgs struct { ... }`.

**Review by chrismwendt — APPROVED** [review]

File: internal/codeintel/uploads/service.go (hunk 1 above)
Comment (Numbers88s) [reply internal/codeintel/uploads/service.go:454]: Went through all of changes on the previous PR. All is well. 👍

**Pull request merged**